WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -0,0 +1,397 @@
+package org.wordpress.android.ui.accounts.login;
+
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.design.widget.TextInputLayout;
+import android.support.v4.app.FragmentTransaction;
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.text.Editable;
+import android.text.TextUtils;
+import android.text.TextWatcher;
+import android.util.Patterns;
+import android.view.KeyEvent;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.BuildConfig;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.generated.AccountActionBuilder;
+import org.wordpress.android.fluxc.store.AccountStore.OnAvailabilityChecked;
+import org.wordpress.android.ui.accounts.AbstractFragment;
+import org.wordpress.android.ui.accounts.SignInDialogFragment;
+import org.wordpress.android.ui.accounts.login.nav.LoginNav;
+import org.wordpress.android.ui.accounts.login.nav.LoginStateGetter;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.AppLog.T;
+import org.wordpress.android.util.EditTextUtils;
+import org.wordpress.android.util.NetworkUtils;
+import org.wordpress.emailchecker2.EmailChecker;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+public class LoginEmailFragment extends AbstractFragment implements TextWatcher {
+    private static final String KEY_IN_PROGRESS = "KEY_IN_PROGRESS";
+
+    public static final String TAG = "login_email_fragment_tag";
+    public static final int MAX_EMAIL_LENGTH = 100;
+
+    protected TextInputLayout mEmailEditTextLayout;
+    protected EditText mEmailEditText;
+
+    protected boolean mEmailAutoCorrected;
+
+    protected Button mNextButton;
+    protected View mUsernamePasswordButton;
+
+    protected @Inject Dispatcher mDispatcher;
```

Comment:
Good point. I've been meaning to cleanup the definitions a bit so, this was a good opportunity. Fixed with 6b6c67d.